lib/ex_hls/demuxing_engine/mpeg_ts.ex

```diff
@@ -80,47 +81,58 @@ defmodule ExHLS.DemuxingEngine.MPEGTS do
 
   @impl true
   def pop_chunk(%__MODULE__{} = demuxing_engine, track_id) do
-    with {[packet], demuxer} <- Demuxer.take(demuxing_engine.demuxer, track_id) do
-      {maybe_tden_tag, demuxer} = maybe_read_tden_tag(demuxer, packet.pts)
+    with {[packet], demuxing_engine} <- take_packets(demuxing_engine, track_id) do
+      {maybe_tden_tag, demuxing_engine} = maybe_read_tden_tag(demuxing_engine, packet.payload.pts)
       tden_tag = maybe_tden_tag || demuxing_engine.last_tden_tag
```

Comment:
for me its unclear if `maybe_read_tden_tag(demuxing_engine, packet.payload.pts)` 
 - returns tden tag
 - or sets it in `demuxing_engine.last_tden_tag`
 - or both

because this function returns `demuxing_engine` as well as `maybe_tden_tag`. What about making it return only `demuxing_engine`?